src/Client/Updater.php

```diff
@@ -118,37 +110,18 @@ class Updater
      *       (the actual update data that has been signed).
      *     - confined_target_dirs: (array) @todo What is this for?
      *       https://github.com/php-tuf/php-tuf/issues/161
-     * @param \ArrayAccess $durableStorage
+     *  @param \Tuf\Metadata\StorageInterface $storage
      *     An implementation of \ArrayAccess that stores its contents durably,
      *     as in to disk or a database. Values written for a given repository
      *     should be exposed to future instantiations of the Updater that
      *     interact with the same repository.
-     *
-     *
      */
-    public function __construct(RepoFileFetcherInterface $repoFileFetcher, array $mirrors, \ArrayAccess $durableStorage)
+    public function __construct(RepoFileFetcherInterface $repoFileFetcher, array $mirrors, StorageInterface $storage)
     {
         $this->repoFileFetcher = $repoFileFetcher;
         $this->mirrors = $mirrors;
-        $this->durableStorage = new DurableStorageAccessValidator($durableStorage);
+        $this->storage = $storage;
         $this->clock = new Clock();
-        $this->metadataFactory = new MetadataFactory($this->durableStorage);
-    }
-
-    /**
-     * Gets the type for the file name.
-     *
-     * @param string $fileName
-     *   The file name.
-     *
-     * @return string
-     *   The type.
-     */
-    private static function getFileNameType(string $fileName): string
```

Comment:
Nice to get rid of this!

Reply:
It was already not in use, so yeah, good time to remove this wart.